managed/models/encryption_helpers.go

```diff
@@ -0,0 +1,296 @@
+	"database/sql"
+	"encoding/json"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/percona/pmm/managed/utils/encryption"
+)
+
+// EncryptAgent encrypt agent.
+func EncryptAgent(agent Agent) Agent {
+	return agentEncryption(agent, encryption.Encrypt)
+}
+
+// DecryptAgent decrypt agent.
+func DecryptAgent(agent Agent) Agent {
+	return agentEncryption(agent, encryption.Decrypt)
+}
+
+func agentEncryption(agent Agent, handler func(string) (string, error)) Agent {
+	if agent.Username != nil {
+		username, err := handler(*agent.Username)
+		if err != nil {
+			logrus.Warning(err)
+		}
+		agent.Username = &username
+	}
+
+	if agent.Password != nil {
+		password, err := handler(*agent.Password)
+		if err != nil {
+			logrus.Warning(err)
+		}
+		agent.Password = &password
+	}
+
+	if agent.AgentPassword != nil {
+		agentPassword, err := handler(*agent.AgentPassword)
+		if err != nil {
+			logrus.Warning(err)
+		}
+		agent.AgentPassword = &agentPassword
+	}
+
+	if agent.AWSAccessKey != nil {
+		awsAccessKey, err := handler(*agent.AWSAccessKey)
+		if err != nil {
+			logrus.Warning(err)
+		}
+		agent.AWSAccessKey = &awsAccessKey
+	}
+
+	if agent.AWSSecretKey != nil {
+		awsSecretKey, err := handler(*agent.AWSSecretKey)
+		if err != nil {
+			logrus.Warning(err)
+		}
+		agent.AWSSecretKey = &awsSecretKey
+	}
+
+	var err error
+	if agent.MySQLOptions != nil {
+		agent.MySQLOptions.TLSCert, err = handler(agent.MySQLOptions.TLSCert)
+		if err != nil {
+			logrus.Warning(err)
+		}
+		agent.MySQLOptions.TLSKey, err = handler(agent.MySQLOptions.TLSKey)
+		if err != nil {
+			logrus.Warning(err)
+		}
+	}
+
+	if agent.PostgreSQLOptions != nil {
+		agent.PostgreSQLOptions.SSLCert, err = handler(agent.PostgreSQLOptions.SSLCert)
+		if err != nil {
+			logrus.Warning(err)
+		}
+		agent.PostgreSQLOptions.SSLKey, err = handler(agent.PostgreSQLOptions.SSLKey)
+		if err != nil {
+			logrus.Warning(err)
+		}
+	}
+
+	if agent.MongoDBOptions != nil {
+		agent.MongoDBOptions.TLSCertificateKey, err = handler(agent.MongoDBOptions.TLSCertificateKey)
+		if err != nil {
+			logrus.Warning(err)
+		}
+		agent.MongoDBOptions.TLSCertificateKeyFilePassword, err = handler(agent.MongoDBOptions.TLSCertificateKeyFilePassword)
+		if err != nil {
+			logrus.Warning(err)
+		}
+	}
+
+	if agent.AzureOptions != nil {
+		agent.AzureOptions.ClientID, err = handler(agent.AzureOptions.ClientID)
+		if err != nil {
+			logrus.Warning(err)
+		}
+		agent.AzureOptions.ClientSecret, err = handler(agent.AzureOptions.ClientSecret)
+		if err != nil {
+			logrus.Warning(err)
+		}
+		agent.AzureOptions.SubscriptionID, err = handler(agent.AzureOptions.SubscriptionID)
+		if err != nil {
+			logrus.Warning(err)
+		}
+		agent.AzureOptions.TenantID, err = handler(agent.AzureOptions.TenantID)
+		if err != nil {
+			logrus.Warning(err)
+		}
+	}
+
+	return agent
+}
+
+// EncryptMySQLOptionsHandler returns encrypted MySQL Options.
+func EncryptMySQLOptionsHandler(e *encryption.Encryption, val any) (any, error) {
```

Comment:
can we change usage of any with some type to make it easier to read

Reply:
Discussed on GO BE.